remote-api/src/main/java/org/alfresco/rest/api/impl/rules/ActionParameterConverter.java

```diff
@@ -159,16 +160,17 @@ private Serializable convertValue(QName typeQName, Object propertyValue) throws
         }
         else
         {
-            if (typeQName.equals(DataTypeDefinition.QNAME) && typeQName.toString().contains(":"))
+            final String stringValue = Objects.toString(propertyValue, null);
+            if (typeQName.isMatch(DataTypeDefinition.QNAME) && typeQName.toString().contains(":"))
             {
-                value = QName.createQName(propertyValue.toString(), namespaceService);
+                value = QName.createQName(stringValue, namespaceService);
             }
             else if (typeQName.isMatch(DataTypeDefinition.NODE_REF))
             {
-                NodeRef nodeRef = nodes.validateOrLookupNode(propertyValue.toString(), null);
+                NodeRef nodeRef = nodes.validateOrLookupNode(stringValue, null);
```

Comment:
This seems a bit weird. For QNames then the method returns `null` but for NodeRefs then it throws an InvalidArgumentException.

Reply:
For the typeQName we have some generic validation that throws a NotFoundException if null is supplied. Maybe we could do something similar (with a different exception) for the propertyValue being null?

Reply:
TBH, I am not sure what to do in case of null property and how to handle it. 
The case here is that even when we get empty String (`""`) as parameter value it is deserialized to `null` so actually we don't know if the client sent a request with null value or empty String value. I'll try to investigate if we can retrieve exactly what was sent.

Reply:
We seem to enforce empty `String`s to be deserialized to `null` by defining `RestApiStringDeserializer` and I understand it is a wider decision that we should put up with.

Reply:
I think we have 2 solutions here that could work in more generic way:
1. Even before we start mapping action parameters - we should completely remove those that have null values (which also means an empty String)
2. Always throw an error on null/empty parameter value (example message: "Action parameter <key> must not be null or empty").
I'd prefer second solution as that way we could "educate" API clients not to send unwanted stuff in requests.
WDYT?